tests/fibonacci.rs

```diff
@@ -48,14 +47,22 @@ fn integration_test() {
     //  ------------ contract data --------------------
 
     let address = Address(1111.into());
-    let class_hash = [1; 32];
-    let contract_state = ContractState::new(class_hash, 3.into(), HashMap::new());
+    let class_hash: ClassHash = [1; 32];
+    let nonce = Felt::zero();
+    let storage_entry: StorageEntry = (address.clone(), [1; 32]);
+    let storage = Felt::zero();
 
     contract_class_cache.insert(class_hash, contract_class);
-    let mut state_reader = InMemoryStateReader::new(HashMap::new(), HashMap::new());
+    let mut state_reader = InMemoryStateReader::default();
     state_reader
-        .contract_states_mut()
-        .insert(address.clone(), contract_state);
+        .address_to_class_hash_mut()
+        .insert(address.clone(), class_hash);
+    state_reader
+        .address_to_nonce_mut()
+        .insert(address.clone(), nonce);
+    state_reader
```

Comment:
same here, can we delete this storage insert?

Reply:
Done